manifests/init.pp

```diff
@@ -11,14 +11,26 @@
   $purge_apt_conf  = false,
 ){
 
-  # Validates that $http_proxy and $http_proxy_port are domain names and ports respectively.
-  if $http_proxy { validate_re($http_proxy, '^([a-z0-9]+(-[a-z0-9]+)*\.)+[a-z]{2,}$') }
-  if $http_proxy_port { validate_re($http_proxy_port, '^\d+$') }
+  # Validates that $http_proxy is a valid, usable domain name or IP address
+  if $http_proxy and size($http_proxy) > 3 {   # '.' is an RFC-valid domain name, but not usable in this context
+    if is_ipv4_address($http_proxy) or is_domain_name($http_proxy) {
+      $_http_proxy = $http_proxy
+    } elsif is_ipv6_address($http_proxy) { # IPv6 needs to be enclosed in []
+      $_http_proxy = enclose_ipv6($http_proxy)
```

Comment:
In the rpm case, rpm.pp creates a file which contains a line for %_httpport and a line for %_httpproxy, and RPM/YUM builds the full proxy RUL on the fly, handling enclosing the IPv6 address in brackets if needed.

For apt configurations, the parameters are passed into the '::apt' class's proxy hash ([package/apt.pp#L9](https://github.com/MiamiOH/puppet-httpproxy/blob/master/manifests/package/apt.pp#L9)) .  In the apt module, they're used to build a single string which requires that the IPv6 address be wrapped in brackets ([init.pp#L76](https://github.com/puppetlabs/puppetlabs-apt/blob/2.2.2/manifests/init.pp#L76)  and [proxy.erb#L1](https://github.com/puppetlabs/puppetlabs-apt/blob/2.2.2/templates/proxy.erb#L1)).  In all likelihood, it's probably more accurate to say that the apt module has a bug - it does not properly handle IPv6 addresses when using them in the template;  this would qualify as a work-around for that bug.